p2p/providers/addrbook_provider/rpc/provider.go

```diff
@@ -0,0 +1,138 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/pokt-network/pocket/p2p/providers/addrbook_provider"
+	"github.com/pokt-network/pocket/p2p/transport"
+	typesP2P "github.com/pokt-network/pocket/p2p/types"
+	"github.com/pokt-network/pocket/rpc"
+	"github.com/pokt-network/pocket/runtime"
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/runtime/defaults"
+	"github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/modules/base_modules"
+)
+
+var (
+	_       addrbook_provider.AddrBookProvider = &rpcAddrBookProvider{}
+	rpcHost string
+)
+
+func init() {
+	// by default, we point at the same endpoint used by the CLI but the debug client is used either in docker-compose of K8S, therefore we cater for overriding
+	rpcHost = runtime.GetEnv("RPC_HOST", defaults.DefaultRPCHost)
+}
+
+type rpcAddrBookProvider struct {
+	base_modules.IntegratableModule
+	base_modules.InterruptableModule
+
+	rpcUrl    string
+	p2pCfg    *configs.P2PConfig
+	rpcClient *rpc.ClientWithResponses
+
+	connFactory typesP2P.ConnectionFactory
+}
+
+func NewRPCAddrBookProvider(options ...modules.ModuleOption) *rpcAddrBookProvider {
+	dabp := &rpcAddrBookProvider{
+		rpcUrl:      fmt.Sprintf("http://%s:%s", rpcHost, defaults.DefaultRPCPort),
+		connFactory: transport.CreateDialer, // default connection factory, overridable with WithConnectionFactory()
+	}
+
+	for _, o := range options {
+		o(dabp)
+	}
+
+	initRPCClient(dabp)
```

Comment:
Make this more functional. One of these two:

1. `dabp.rpcClient = initRPCClient(dabp.rpcUrl)`
2. `dabp.initRPCClient()`